src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/BuildJobExecutionService.java

```diff
@@ -398,14 +397,18 @@ private void processTestCaseNode(XMLStreamReader xmlStreamReader, List<LocalCIBu
         // Check if there is a failure node inside the testcase node.
         // Call next() until there is an end element (no failure node exists inside the testcase node) or a start element (failure node exists inside the
         // testcase node).
-        xmlStreamReader.next();
-        while (!(xmlStreamReader.isEndElement() || xmlStreamReader.isStartElement())) {
+        do {
             xmlStreamReader.next();
         }
+        while (!(xmlStreamReader.isEndElement() || xmlStreamReader.isStartElement()));
         if (xmlStreamReader.isStartElement() && "failure".equals(xmlStreamReader.getLocalName())) {
             // Extract the message attribute from the "failure" node.
             String error = xmlStreamReader.getAttributeValue(null, "message");
 
+            if (error == null && xmlStreamReader.hasNext()) {
```

Comment:
Failures and errors can contain both the message attribute and inner text, which should be combined (probably with a newline)

Reply:
I'll have another look at the format in detail and will fix that in a followup.